### ISSUE TITLE
Favorite / Pinned Channels

### ISSUE THREAD

**Issue by arche-dev** [ajbura/cinny#44]
#### Is your feature request related to a problem? Please describe.

Currently there's no favorite channel/room section

#### Describe the solution you'd like

Add a section for favorite channels, this will also require an option to pin/favorite channels in the options. (Maybe make it called pinned channels? Same-ish issue wih #30)

**Comment by farribeiro** [comment]
This is a HS feature.

**Comment by BBaoVanC** [comment]
> This is a HS feature.

Which homeservers don't support it?

**Comment by arche-dev** [comment]
> > This is a HS feature.
> 
> Which homeservers don't support it?

I'm pretty sure ever HS supports it

**Comment by hifi** [comment]
Would be nice if the existing pins that other clients already support (Element and FluffyChat both share the pinned room option) would be honored even without any UI for setting them.

**Comment by pinpox** [comment]
+1 for this feature. I have a few channels/rooms I use regularly and a long list that I just lurk in. Searching for the channels all the time is annoying, I'd like to have a few ones at the top constantly

**Comment by ginnyTheCat** [comment]
I created a kind of favorites space, added all the channels I often use to and pinned the space to the sidebar. It's a kind of workaround but it's ok.

**Comment by pinpox** [comment]
> I created a kind of favorites space, added all the channels I often use to and pinned the space to the sidebar. It's a kind of workaround but it's ok.

Yeah, thought about that workaround too, but it gets messy if you have other clients (e.g. on mobile devices) that have the pinned feature, as those are synced server-side if I'm not mistaken. 

In any case, I think this feature would be a great addition to cinny and probably not too unreasonable to implement for someone that knows the codebase.

**Comment by IngwiePhoenix** [comment]
Basically the only feature missing for me, everything else's there already!

**Comment by pinpox** [comment]
Any updates on this?

**Comment by u9000-Nine** [comment]
So my concern is that if you're in a lot of spaces, there are already a ton of icons on the sidebar. I'd really like a solution though that allows you to pin rooms in the Home page, and pin DMs in the DMs page. That would basically make spaces, miscellaneous rooms, and DMs all rearrangeable to some degree.

**Comment by pinpox** [comment]
Implemented in #2977
